src/include/simeng/OS/Vma.hh

```diff
@@ -1,50 +1,163 @@
 #pragma once
 
 #include <stdint.h>
+#include <sys/mman.h>
 
 #include <cstddef>
 #include <iostream>
 #include <memory>
+#include <vector>
 
+#include "simeng/OS/Constants.hh"
 #include "simeng/util/Math.hh"
+
 namespace simeng {
 namespace OS {
 
-/** struct representing vm_area_struct in linux. (Linked List) */
-struct VirtMemArea {
+using namespace simeng::OS::defaults;
+
+/** A HostFileMMap is a structure representing a file mmaped on the host. */
+class HostFileMMap {
+ public:
+  /** Host fd of the file mmaped by host. */
+  const int fd_;
+
+  /** Length of the file mapping after offset has been applied. When a mmap call
+   * is made it can contain an offset and the offset has to be a multiple of
+   * page size. To ensure SimEng doesn't run into alignment conflicts due to
+   * different page sizes on different systems. Whenever a mmap call is made
+   * with a file descriptor and offset. SimEng mmaps the entire file on the host
+   * system and calculates the effective length given the original file length
+   * and offset i.e original_file_length - offset. */
```

Comment:
This comment is good, but the punctuation makes it hard to read. Could you clean it up a bit please